src/components/Modal/Modal.jsx

```diff
@@ -18,32 +21,27 @@ const preRender = (
   restProps,
   size,
 ) => (
-  <div
-    className={styles.backdrop}
+  <dialog
+    {...transferProps(restProps)}
+    className={classNames(
+      styles.root,
+      getSizeClassName(size, styles),
+      getPositionClassName(position, styles),
+    )}
     onClick={(e) => {
+      e.stopPropagation();
+    }}
+    onClose={(e) => {
       e.preventDefault();
       if (closeButtonRef?.current != null) {
         closeButtonRef.current.click();
       }
     }}
+    ref={childrenWrapperRef}
```

Comment:
A reference to the top-level HTML element can be simply passed through `restProps`, is it correct? I therefore think the `childrenWrapperRef` prop can be removed.

Reply:
Oh, I see we use the reference in a hook… I'm eager to learn new things then 🙂.

Reply:
`childrenWrapperRef` is part of `preRender` function used by `Modal` component. As it is passed as `ref`, it cannot be send to the `preRender` function just via rest props.. `childrenWrapperRef` must be there to call `showModal()` to show modal dialog (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog).